### PR TITLE
syslog-ng-otlp(): only send tags that have global relevance (e.g. not starting with a dot)

### PR DESCRIPTION
**Pull request by bazsi**
This branch starts to differentiate between global tags and local tags. Local tags start with a '.' and all previous built-in use of tags will be considered local).

The $TAGS macro would continue to expand to everything, so we would remain compatible, but the syslog-ng-otlp() driver is changed so that only global tags are transmitted to the peer.

This also contains a change to the $PRI value which is becoming a simple integer instead of a string representation of the same.

### CODE REVIEW

File: modules/grpc/otel/otel-protobuf-parser.cpp [hunk 1]
```diff
@@ -1209,9 +1209,9 @@ syslogng::grpc::otel::ProtobufParser::set_syslog_ng_macros(LogMessage *msg, cons
 
       if (name.compare("PRI") == 0)
         {
-          if (!_value_case_equals(msg, macro, AnyValue::kBytesValue))
+          if (!_value_case_equals(msg, macro, AnyValue::kIntValue))
             continue;
-          msg->pri = log_rewrite_set_pri_convert_pri(macro.value().bytes_value().c_str());
+          msg->pri = macro.value().int_value();
```
Comment (alltilla) [inline modules/grpc/otel/otel-protobuf-parser.cpp:1230]: I think we should keep the bytes value parsing, too, for the sake of backward compatibility. We can check for int first, as it is the most probable.

(e.g. syslog-ng 4.6 sender -> syslog-ng 4.7 receiver)

File: modules/grpc/otel/otel-protobuf-formatter.cpp [hunk 2]
```diff
@@ -584,18 +584,16 @@ ProtobufFormatter::set_syslog_ng_macros(LogMessage *msg, LogRecord &log_record)
...
-  LogMessageValueType type;
-  gssize len;
-  const char *value;
-
   KeyValue *pri_attr = macros_kvlist->add_values();
   pri_attr->set_key("PRI");
   pri_attr->mutable_value()->set_int_value(msg->pri);
 
-  value = log_msg_get_value_by_name_with_type(msg, "TAGS", &len, &type);
+  GString *tags_value = g_string_sized_new(64);
```
Comment (alltilla) [inline modules/grpc/otel/otel-protobuf-formatter.cpp:591]: [optional]

I know that gRPC already has a big memory allocation overhead, but I would try to keep the memory allocations minimal where we are able to.

In this case I would either introduce scratch buffers on the OTLP protobuf formatting side, or I would add a dedicated macro for global tags (`$GLOBAL_TAGS` ?) and use `log_msg_get_value_by_name_with_type()` here, and let `LogMessage::log_msg_get_macro_value()`'s `value = g_private_get(&priv_macro_value);` optimization take care of it.

File: lib/ivykis
Comment (alltilla) [inline lib/ivykis:1]: Is this an intentional ivykis submodule update?

File: modules/grpc/otel/otel-protobuf-parser.cpp [hunk 3]
```diff
@@ -1091,10 +1110,7 @@ syslogng::grpc::otel::ProtobufParser::store_raw_metadata(LogMessage *msg, const
 {
   std::string serialized;
 
-  /* HOST */
-  std::string hostname = _extract_hostname(peer);
-  if (hostname.length())
-    log_msg_set_value(msg, LM_V_HOST, hostname.c_str(), hostname.length());
+  msg->saddr = _extract_saddr(peer);
```
Comment (alltilla) [inline modules/grpc/otel/otel-protobuf-parser.cpp:1113]: After this change the `$HOST` macro will always be empty. Isn't this a problem? Shouldn't we store this value in `$HOST` as well?

File: modules/grpc/otel/otel-protobuf-formatter.cpp (hunk 2 above)
Comment (bazsi) [reply modules/grpc/otel/otel-protobuf-formatter.cpp:591]: I don't want to use the by_name() versions of those functions, that means an extra lookup (name->handle->nvtable)

**Comment by MrAnno** [comment]
I've fixed the review comments and removed the ivykis and host changes.
I'm going to open another PR about improving how we set HOST, HOST_FROM in opentelemetry() and syslog-ng-otlp().

**Review by MrAnno — APPROVED** [review]

**Pull request merged**